### PR TITLE
uarmsolver: init at 0.2.4

### PR DESCRIPTION
**Pull request by firefly-cpp**
###### Description of changes

[uARMSolver](https://github.com/firefly-cpp/uARMSolver) is a universal Association Rule Mining Solver.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by SuperSandro2000** [comment]
Please add yourself as a maintainer by adding an alphabetically sorted entry to `maintainers/maintainer-list.nix` and commit this change into a separate commit with the message ``maintainers: add NAME`` into this PR.

File: pkgs/applications/science/machine-learning/uarmsolver/default.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+   owner = "firefly-cpp";
+   repo = "uARMSolver";
+   rev = "${version}";
+   sha256 = "17px69z0kw0z6cip41c45i6srbw56b0md92i9vbqyzinx8b75mzw";
+  };
+
+  nativeBuildInputs = [ cmake ];
+
+  installPhase = ''
+   install -D -t $out/bin uARMSolver
```
Comment (SuperSandro2000) [inline pkgs/applications/science/machine-learning/uarmsolver/default.nix:21]:
```suggestion
    install -D -t $out/bin uARMSolver
```

File: pkgs/applications/science/machine-learning/uarmsolver/default.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+  meta = with lib; {
+   description = "universal Association Rule Mining Solver";
+   homepage    = "https://github.com/firefly-cpp/uARMSolver";
+   license     = licenses.mit;
+   platforms   = platforms.all;
+   maintainers = with maintainers; [ firefly-cpp ];
+  };
+
+}
```
Comment (SuperSandro2000) [inline pkgs/applications/science/machine-learning/uarmsolver/default.nix:32]:
```suggestion
  meta = with lib; {
    description = "universal Association Rule Mining Solver";
    homepage    = "https://github.com/firefly-cpp/uARMSolver";
    license     = licenses.mit;
    platforms   = platforms.all;
    maintainers = with maintainers; [ firefly-cpp ];
  };
}
```

File: pkgs/applications/science/machine-learning/uarmsolver/default.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  name = "uarmsolver";
+  version = "0.2.4";
+
+  src = fetchFromGitHub {
+   owner = "firefly-cpp";
+   repo = "uARMSolver";
+   rev = "${version}";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/machine-learning/uarmsolver/default.nix:14]:
```suggestion
   rev = version;
```

**Comment by firefly-cpp** [comment]
Thanks @SuperSandro2000 for your suggestions!

**Comment by SuperSandro2000** [comment]
You reverted the changes in the force push.

**Comment by firefly-cpp** [comment]
Corrected now. Thanks.

**Comment by firefly-cpp** [comment]
@SuperSandro2000, are there any other problems with these commits? Do we need to squash commits?

File: pkgs/applications/science/machine-learning/uarmsolver/default.nix [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+}:
+
+stdenv.mkDerivation rec {
+  name = "uarmsolver";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/machine-learning/uarmsolver/default.nix:8]:
```suggestion
  pname = "uarmsolver";
```

File: pkgs/applications/science/machine-learning/uarmsolver/default.nix [hunk 5]
```diff
@@ -0,0 +1,31 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+}:
+
+stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/science/machine-learning/uarmsolver/default.nix:7]:
```suggestion
gccStdenv.mkDerivation rec {
```

File: pkgs/applications/science/machine-learning/uarmsolver/default.nix [hunk 6]
```diff
@@ -0,0 +1,31 @@
+{ lib
+, stdenv
```
Comment (SuperSandro2000) [inline pkgs/applications/science/machine-learning/uarmsolver/default.nix:2]:
```suggestion
, gccStdenv
```

File: pkgs/applications/science/machine-learning/uarmsolver/default.nix [hunk 7]
```diff
@@ -0,0 +1,31 @@
+{ lib
+, gccStdenv
```
Comment (SuperSandro2000) [inline pkgs/applications/science/machine-learning/uarmsolver/default.nix:2]:
```suggestion
, stdenv
```

File: pkgs/applications/science/machine-learning/uarmsolver/default.nix [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+, fetchFromGitHub
+, cmake
+}:
+
+gccStdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/applications/science/machine-learning/uarmsolver/default.nix:7]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/applications/science/machine-learning/uarmsolver/default.nix [hunk 9]
```diff
@@ -0,0 +1,31 @@
...
+
+  installPhase = ''
+    install -D -t $out/bin uARMSolver
+  '';
+
+  meta = with lib; {
+    description = "universal Association Rule Mining Solver";
+    homepage    = "https://github.com/firefly-cpp/uARMSolver";
+    license     = licenses.mit;
+    platforms   = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/applications/science/machine-learning/uarmsolver/default.nix:28]:
```suggestion
    platforms   = platforms.linux;
```

**Pull request merged**